src/app/search-page/search.component.ts

```diff
@@ -0,0 +1,253 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { PageEvent } from '@angular/material/paginator';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subject, Subscription, takeUntil } from 'rxjs';
+import { ProductService } from '../api/products.service';
+import { QueryService } from '../api/query.service';
+import { IProduct } from '../Interface/products';
+
+
+@Component({
+  selector: 'app-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.scss'],
+})
+
+export class SearchComponent implements OnInit, OnDestroy {
+  query: string | null = '';
+
+  allProductsBySearch: IProduct[] = [];
+
+  filteredProducts: IProduct[] = [];
+
+  searchCount: number | undefined;
+
+  sub!: Subscription;
+
+  visibleProducts: IProduct[] = [];
+
+  categoryList: string[] = [];
+
+  statusList: string[] = [];
+
+  releaseDateList: string[] = ['Newest', 'Oldest'];
+
+  categories = new FormControl();
+
+  status = new FormControl();
+
+  releaseDate = new FormControl();
+
+  private _selectedCategory: string = '';
+
+  private _selectedStatus: string = '';
+
+  private _selectedReleaseDate: string = '';
+
+  private readonly destroy$ = new Subject<void>();
+
+  private _searchDetails: string = '';
+
+  breakPoint!: number;
+
+  constructor(private route: ActivatedRoute,
+    private router: Router,
+    private productService: ProductService,
+    private queryService: QueryService) { }
+
+  get searchDetails(): string {
+    return this._searchDetails;
+  }
+
+  set searchDetails(value: string) {
+    this._searchDetails = value;
+  }
+
+  set selectedCategory(value: string) {
+    this.resetFilterValues();
+    this._selectedCategory = value;
+
+    this.filteredProducts = findItemsByCategory(this._selectedCategory, this.allProductsBySearch);
+    this.visibleProducts = this.filteredProducts.slice(0, 12);
+  }
+
+  get selectedCategory() {
+    return this._selectedCategory;
+  }
+
+  set selectedStatus(value: string) {
+    this.resetFilterValues();
+    this._selectedStatus = value;
+
+    this.filteredProducts = findItemsByStatus(this._selectedStatus, this.allProductsBySearch);
+    this.visibleProducts = this.filteredProducts.slice(0, 12);
+  }
+
+  get selectedStatus() {
+    return this._selectedStatus;
+  }
+
+  set selectedReleaseDate(value: string) {
+    this._selectedReleaseDate = value;
+
+    if (this.filteredProducts.length === 0) {
+      sortProductsByReleaseDate(this.allProductsBySearch, value);
+      this.visibleProducts = this.allProductsBySearch.slice(0, 12);
+    }
+
+    if (this.filteredProducts.length > 0) {
+      sortProductsByReleaseDate(this.filteredProducts, value);
+      this.visibleProducts = this.filteredProducts.slice(0, 12);
+    }
+  }
+
+  get selectedReleaseDate() {
+    return this._selectedReleaseDate;
+  }
+
+  onResize(event: any) {
+    this.breakPoint = (event.target.innerWidth < 1200) ? 2 : 3;
```

Comment:
Why is this 1200 here and 1000 in ngOnInit? :)

Reply:
:)) changed it to 1200 in ngOnInit